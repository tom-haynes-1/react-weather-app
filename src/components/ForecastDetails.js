import React from "react";
import PropTypes from "prop-types";

function ForecastDetails({ forecast }) {
  const { date, temperature, humidity, wind } = forecast;
  const formattedDate = new Date(date).toDateString();

  return (
    <div className="forecast-details" data-testid="forecast-details">
      <div className="forecast-details__date">{formattedDate}</div>
      <div className="forecast-details__maxTemperature">
        Max Temperature: {temperature.max}&deg;C
      </div>
      <div className="forecast-details__minTemperature">
        Min Temperature: {temperature.min}&deg;C
      </div>
      <div className="forecast-details__humidity">Humidity: {humidity}%</div>
      <div className="forecast-details__wind_speed">
        Wind Speed: {wind.speed}mph
      </div>
      <div className="forecast-details__wind_direction">
        Wind Direction: {wind.direction}
      </div>
    </div>
  );
}

export default ForecastDetails;

ForecastDetails.propTypes = {
  forecast: PropTypes.shape({
    date: PropTypes.number,
    temperature: PropTypes.shape({
      max: PropTypes.number,
      min: PropTypes.number,
    }),
    wind: PropTypes.shape({
      speed: PropTypes.number,
      direction: PropTypes.string,
    }),
    humidity: PropTypes.number,
  }).isRequired,
};
