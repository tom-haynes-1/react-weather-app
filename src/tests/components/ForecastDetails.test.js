/* eslint-disable prettier/prettier */
import React from "react";
import { render } from "@testing-library/react";
import ForecastDetails from "../../components/ForecastDetails";

describe("ForecastDetails", () => {
  const forecast = {
    date: 1525046400000,
    temperature: {
      max: 11,
      min: 4,
    },
    humidity: 40,
    wind: {
      speed: 60,
      direction: "sw",
    },
  };

  it("renders correctly", () => {
    const { asFragment } = render(<ForecastDetails forecast={forecast} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders correct values for props", () => {
    const { getByText } = render(<ForecastDetails forecast={forecast} />);

    expect(getByText("Mon Apr 30 2018")).toHaveClass("forecast-details__date");
    expect(getByText("Max Temperature: 11°C")).toHaveClass("forecast-details__maxTemperature");
    expect(getByText("Min Temperature: 4°C")).toHaveClass("forecast-details__minTemperature");
    expect(getByText("Humidity: 40%")).toHaveClass("forecast-details__humidity");
    expect(getByText("Wind Speed: 60mph")).toHaveClass("forecast-details__wind_speed");
    expect(getByText("Wind Direction: sw")).toHaveClass("forecast-details__wind_direction");
  });
});
