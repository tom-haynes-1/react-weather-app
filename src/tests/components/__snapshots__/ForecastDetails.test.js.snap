// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDetails renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-details"
    data-testid="forecast-details"
  >
    <div
      class="forecast-details__date"
    >
      Mon Apr 30 2018
    </div>
    <div
      class="forecast-details__maxTemperature"
    >
      Max Temperature: 11°C
    </div>
    <div
      class="forecast-details__minTemperature"
    >
      Min Temperature: 4°C
    </div>
    <div
      class="forecast-details__humidity"
    >
      Humidity: 40%
    </div>
    <div
      class="forecast-details__wind_speed"
    >
      Wind Speed: 60mph
    </div>
    <div
      class="forecast-details__wind_direction"
    >
      Wind Direction: sw
    </div>
  </div>
</DocumentFragment>
`;
